core/src/cluster_slots.rs

```diff
@@ -40,6 +44,8 @@ impl ClusterSlots {
     }
 
     fn update_internal(&self, root: Slot, epoch_slots_list: Vec<EpochSlots>) {
+        // Discard slots at or before current root or epochs ahead.
+        const SLOTS_DIFF_RANGE: Range<u64> = 1..2 * DEFAULT_SLOTS_PER_EPOCH;
```

Comment:
Ideally we get this from the runtime value: `bank.get_slots_in_epoch()`

Reply:
done.
added an extra `.max(DEFAULT_SLOTS_PER_EPOCH)` just to be a bit more conservative in discarding epoch-slots.